pkg/rasactl/status.go

```diff
@@ -20,33 +20,71 @@ import (
 
 	"github.com/RasaHQ/rasactl/pkg/status"
 	"github.com/RasaHQ/rasactl/pkg/types"
+	"helm.sh/helm/v3/pkg/release"
 )
 
-// Status prints status for a given deployment.
-func (r *RasaCtl) Status() error {
-	var d = [][]string{}
+type StatusProject string
+
+const (
+	StatusStopped    StatusProject = "Stopped"
+	StatusRunning    StatusProject = "Running"
+	StatusInstalling StatusProject = "Installing"
+	StatusUpgrading  StatusProject = "Upgrading"
+)
+
+// Return project status, helm release, and err for a given helm release
+func (r *RasaCtl) GetReleaseStatus(releaseName []byte) (StatusProject, *release.Release, error) {
+
+	status := StatusStopped
+
 	isRunning, err := r.KubernetesClient.IsRasaXRunning()
 	if err != nil {
-		return err
+		return status, nil, err
 	}
-	statusProject := "Stopped"
+
 	if isRunning {
-		statusProject = "Running"
+		status = StatusRunning
 	}
 
-	stateData, err := r.KubernetesClient.ReadSecretWithState()
-	if err != nil {
-		return err
-	}
 	r.HelmClient.SetConfiguration(
 		&types.HelmConfigurationSpec{
-			ReleaseName: string(stateData[types.StateHelmReleaseName]),
+			ReleaseName: string(releaseName),
 		},
 	)
-	r.KubernetesClient.SetHelmReleaseName(string(stateData[types.StateHelmReleaseName]))
+	r.KubernetesClient.SetHelmReleaseName(string(releaseName))
+
+	release, err := r.HelmClient.GetStatus()
+
+	if err != nil {
+		return status, release, err
+	}
+
+	statusRelease := release.Info.Status.String()
+	if statusRelease == "pending-upgrade" {
```

Comment:
I'd use `switch` for it. I think it'll be more readable.

Reply:
Updated